### PR TITLE
Add __traits(classInstanceAlignment)

### PR DESCRIPTION
**Pull request by kinke**
For https://github.com/dlang/dmd/pull/14063.

### CODE REVIEW

**Comment by kinke** [comment]
(The Phobos library trait is buggy: https://g.godbolt.org/z/c4MWov45f)

File: spec/traits.dd [hunk 1]
```diff
@@ -580,6 +581,17 @@ $(H3 $(GNAME classInstanceSize))
...
+$(H3 $(GNAME classInstanceAlignment))
+
+        $(P Takes a single argument, which must evaluate to either
+        a class type or an expression of class type.
+        The result
+        is of type $(CODE size_t), and the value is the alignment
+        of a runtime instance of the class type.
+        It is based on the static type of a class, not the
+        polymorphic type.
+        )
```
Comment (ljmf00) [inline spec/traits.dd:593]: I would suggest 80 char text wrapping.

```suggestion
        $(P Takes a single argument, which must evaluate to either a class type
        or an expression of class type.
        The result is of type $(CODE size_t), and the value is the alignment of
        a runtime instance of the class type. It is based on the static type of
        a class, not the polymorphic type.
        )
```

Comment (kinke) [reply spec/traits.dd:593]: [The block is a copy-pasta from classInstanceSize above.]

Comment (ljmf00) [reply spec/traits.dd:593]: Right, we can refactor those in a subsequent PR.

**Review by ljmf00 — APPROVED** [review]
LGTM, although I don't have approval rights

File: spec/traits.dd [hunk 2]
```diff
@@ -580,6 +581,17 @@ $(H3 $(GNAME classInstanceSize))
         polymorphic type.
         )
 
+$(H3 $(GNAME classInstanceAlignment))
+
+        $(P Takes a single argument, which must evaluate to either
+        a class type or an expression of class type.
+        The result
+        is of type $(CODE size_t), and the value is the alignment
+        of a runtime instance of the class type.
```
Comment (maxhaton) [inline spec/traits.dd:590]:
```suggestion
        of an instance of the class type.
```
Runtime is probably unnecessary here

File: spec/traits.dd [hunk 3]
```diff
@@ -580,6 +581,17 @@ $(H3 $(GNAME classInstanceSize))
...
+        It is based on the static type of a class, not the
```
Comment (maxhaton) [inline spec/traits.dd:591]: Is dynamic type a concept we use in the spec elsewhere, it's quite clunky

**Review by maxhaton — APPROVED** [review]

**Pull request merged**